plugins/inputs/lustre2/lustre2.go

```diff
@@ -432,6 +455,90 @@ func (l *Lustre2) GetLustreProcStats(fileglob string, wantedFields []*mapping) e
 	return nil
 }
 
+func (l *Lustre2) GetLustreProcBrwStats(fileglob string, wantedFields []*mapping) error {
+	files, err := filepath.Glob(fileglob)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		// Turn /proc/fs/lustre/obdfilter/<ost_name>/stats and similar into just the object store target name
+		// This assumes that the target name is always second to last, which is true in Lustre 2.1->2.12
+		path := strings.Split(file, "/")
+		name := path[len(path)-2]
+
+		wholeFile, err := ioutil.ReadFile(file)
+		if err != nil {
+			return err
+		}
+		lines := strings.Split(string(wholeFile), "\n")
+
+		var header_name string
+
+		for _, line := range lines {
+			// There are four types of lines in a brw_stats file:
+			// 1. Header lines - contain the category of metric (e.g. disk I/Os in flight, disk I/O time)
+			// 2. Bucket lines - follow headers, contain the bucket value (e.g. 4K, 1M) and metric values
+			// 3. Empty lines - these will simply be filtered out
+			// 4. snapshot_time line - this will be filtered out, as it "looks" like a bucket line
+			if len(line) < 1 {
+				continue
+			}
+			parts := strings.Fields(line)
+
+			if strings.Contains(parts[0], ":") == false { // This is a header line
+				nameParts := strings.Split(line, "  ")
+				header_name = nameParts[0] // Set report name for use by the buckets that follow
+				continue
+			} else if strings.Contains(parts[0], "snapshot_time") { // snapshot_time should be discarded
+				continue
+			} else { // This is a bucket for a given header
+				for _, wanted := range wantedFields {
+					if header_name == wanted.inProc {
```

Comment:
If you invert the check here, you can save another level of indentation.
```suggestion
					if header_name != wanted.inProc {
						continue
					}
```